java/de/yessoft/android/mohammad/ui/MainActivity/MainActivity.java

```diff
@@ -0,0 +1,81 @@
+package de.yessoft.android.mohammad.ui.MainActivity;
+
+import android.os.Bundle;
+import android.widget.Toast;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.firebase.ui.auth.AuthUI;
+import com.google.firebase.auth.FirebaseAuth;
+import com.google.firebase.auth.FirebaseUser;
+
+import java.util.Arrays;
+import java.util.List;
+
+import de.yessoft.android.mohammad.R;
+import de.yessoft.android.mohammad.ui.MainActivity.fragments.NewsFeedFragment;
+
+public class MainActivity extends AppCompatActivity {
+
+    // private static final String TAG = "ML_MainActivity";
+    private static final int RC_SIGN_IN = 19;
+
+    // region Life Cycle of an Activity
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        // region Check Login Status and Redirect If Not Logged In
+        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
```

Comment:
MainActivity  is view correct ? 

and here I can see firebase , I think you got my point .

Activity should not know anything about firbase

Reply:
No, it's a controller, i use fragments as a view